src/Ternel/Weekend.php

```diff
@@ -157,4 +161,63 @@ private function checkTomorrowNotWorkingDay()
 
         return array_search($tomorrow, $this->getHolidays());
     }
+
+    /**
+     * Avoid calling ip file every day & every hit
+     *
+     * @param $easterDate
+     * @param $year
+     * @return bool
+     */
+    private function isAlsacianHolidaysDate($easterDate, $year)
+    {
+        //3 days before easter, can't $easterDay - 2 if easter is the first day of month
+         return date('d-m-Y') == date('d-m-Y', mktime(0, 0, 0, 12, 26, $year)) ||
+            date('d-m-Y') == date('d-m-Y', mktime($easterDate - (3600 * 2 * 24)));
+    }
+
+    /**
+     * We call ipinfo if we are on alsacian holidays
+     *
+     * @param $easterDate
+     * @param $year
+     * @return array
+     */
+    private function addAlsacianHolidays($easterDate, $year)
+    {
+        //3 days before easter, can't $easterDay - 2 if easter is the first day of month
+        $alsacianHolidays = [
+            'vendredisaint' => date('d-m-Y', mktime($easterDate - (3600 * 2 * 24))),
+            'saintetienne' => date('d-m-Y', mktime(0, 0, 0, 12, 26, $year))
+        ];
+
+        //fast ip
+        if (isset($_SERVER['HTTP_X_FORWARDED_FOR'])) {
+            $ip = $_SERVER['HTTP_X_FORWARDED_FOR'];
+        }elseif (isset($_SERVER['HTTP_CLIENT_IP'])) {
+            $ip = $_SERVER['HTTP_CLIENT_IP'];
+        }else{
+            $ip = $_SERVER['REMOTE_ADDR'];
+        }
+
+        //we allow only 3 districts
+        $districsAllowed = [
+            '67',
+            '68',
+            '90',
+            '57'
+        ];
+
+        //yolo json
+        $json = json_decode(file_get_contents('http://ipinfo.io/' . $ip . '/json'));
```

Comment:
You should add a try/catch around this because the rate limite is 1,000 Daily Requests, and this website is so popular it might break really really fast :+1: